### PR TITLE
fix: [PIE-5415]: Multi Service - UI - Bug Fixes

### PR DESCRIPTION
**Pull request by vkbansal-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 3 Issues 
* Code issue 3 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11319)

File: src/modules/10-common/components/MultiTypeMultiSelectDropDown/MultiTypeMultiSelectDropDown.tsx [hunk 1]
```diff
@@ -29,6 +32,38 @@ import { useStrings } from 'framework/strings'
...
+
+export function MultiSelectDropDownFixed(props: MultiSelectDropDownFixedProps): React.ReactElement {
+  const { onChange, value, disabled, ...rest } = props
+  const [localValue, setLocalValue] = useState<MultiSelectOption[]>([])
+
+  useEffect(() => {
+    if (Array.isArray(value)) {
+      setLocalValue(value)
+    }
+  }, [value])
```
Comment (udayvunnam) [inline src/modules/10-common/components/MultiTypeMultiSelectDropDown/MultiTypeMultiSelectDropDown.tsx:45]: Why an effect than an inline/useMemo? With we are deferring the update to the next render

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/10-common/components/MultiTypeMultiSelectDropDown/MultiTypeMultiSelectDropDown.tsx [hunk 2]
```diff
@@ -29,6 +32,38 @@ import { useStrings } from 'framework/strings'
...
+    onChange?.(opts as SelectOption[], MultiTypeInputValue.MULTI_SELECT_OPTION, MultiTypeInputType.FIXED)
+    rest.onPopoverClose?.(opts)
+  }
+
+  return (
+    <MultiSelectDropDown
+      {...rest}
+      value={localValue}
+      disabled={disabled}
+      onChange={handleChangeNoop}
```
Comment (ishantmehtaharness) [inline src/modules/10-common/components/MultiTypeMultiSelectDropDown/MultiTypeMultiSelectDropDown.tsx:61]: onChange={setLocalValue}

Comment (vkbansal-harness) [reply src/modules/10-common/components/MultiTypeMultiSelectDropDown/MultiTypeMultiSelectDropDown.tsx:61]: more changes might be required in future

**Review by ishantmehtaharness — APPROVED** [review]

**Pull request merged**